src/Auth/DatabaseAuthenticator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Auth;
+
+use Tempest\Auth\Exceptions\InvalidLoginException;
+use Tempest\Database\Query;
+
+final class DatabaseAuthenticator extends GenericAuthenticator
+{
+    /**
+     * @throws InvalidLoginException
+     */
+    public function login(Identifiable $identifiable): void
+    {
+        if (! $this->identifiableExists($identifiable)) {
+            throw new InvalidLoginException();
+        }
+
+        $this->createSession($identifiable);
+    }
+
+    /**
+     * @return Identifiable|null
+     */
+    public function user()
```

Comment:
@brendt the idea here is to return the class that the user implements the `Identifiable` interface, but IDK the best approach to map the query result to the target class, do you have any ideas on this?

Reply:
I'm not sure I understand the question? Are you wondering about how to store it in the session?

Reply:
No, I was thinking how I could return the model here instead of the array but I have an ideia, I’ll implement it so you can review